src/routes/project/TaskRow.svelte

```diff
@@ -22,11 +24,35 @@
 		inProgress: 'task--inProgress',
 		todo: 'task--todo',
 	}
+
+	let chartNode: HTMLElement
+	let isVisible: boolean = false
+	let rightPosition: boolean = false
+
+	$: {
+		if (chartNode instanceof HTMLElement) {
+			const taskLeftSide = chartNode.offsetLeft
+			const taskRightSide = chartNode.offsetLeft + chartNode.offsetWidth
+			isVisible = taskLeftSide >= leftBorder && taskRightSide <= rightBorder
+			if (taskRightSide > rightBorder) {
+				rightPosition = true
+			} else {
+				rightPosition = false
+			}
+		}
+	}
 </script>
 
 <div class="task {taskStatusClasses[task.status]}" style="opacity: {task.isDormant ? 0.3 : 1};">
-	<div class="task__chart" style="margin-left: {10 * task.waitTime}px">
-		<div class="task__detail-name">{task.name}</div>
+	<div class="task__chart" style="margin-left: {10 * task.waitTime}px" bind:this={chartNode}>
```

Comment:
Now, when we have a permanently attached node with title, we can measure it, instead of this one.